router/analytics.go

```diff
@@ -56,6 +57,12 @@ func (a *Analytics) run() {
 			continue
 		}
 
-		_, _ = http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		resp, err := http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		if err != nil && resp != nil && resp.Body != nil {
+			// read body to end and close it; otherwise we
+			// leak some resources in the http transport
+			_, _ = ioutil.ReadAll(resp.Body)
```

Comment:
actually, Close() drains data automatically. So maybe we can remove ReadAll here?

Reply:
Didn't know that, thanks.  When is doEarlyClose set/unset?  (also, ioutil.Discard seems like a better way to write this)

Reply:
It happens at the server side for connection reuse. https://github.com/golang/go/blob/2fd97ee25e8a623e460aff26b4c422dec9532654/src/net/http/server.go#L1007-L1009